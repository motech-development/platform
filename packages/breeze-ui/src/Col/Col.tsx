import React, { FC, memo, ReactNode } from 'react';
import styled from 'styled-components';

interface IColContainer {
  align: 'centre' | 'left' | 'right';
  lg: number;
  lgOffset: number;
  md: number;
  mdOffset: number;
  sm: number;
  smOffset: number;
  verticalAlign: boolean;
  xs: number;
  xsOffset: number;
}

const ColContainer = styled.div<IColContainer>`
  ${({
    align,
    lg,
    lgOffset,
    md,
    mdOffset,
    sm,
    smOffset,
    verticalAlign,
    xs,
    xsOffset,
  }) => `
    grid-column: ${xsOffset > 0 ? `${xsOffset} / span ${xs}` : `span ${xs}`};
    position: relative;
    text-align ${align === 'centre' ? 'center' : align};

    ${
      verticalAlign
        ? `
      align-items: center;
      display: flex;
    `
        : ``
    }

    ${
      sm
        ? `
      @media (min-width: 576px) {
        grid-column: ${
          smOffset > 0 ? `${smOffset} / span ${sm}` : `span ${sm}`
        };
      }
    `
        : ''
    }

    ${
      md
        ? `
      @media (min-width: 768px) {
        grid-column: ${
          mdOffset > 0 ? `${mdOffset} / span ${md}` : `span ${md}`
        };
      }
    `
        : ''
    }

    ${
      lg
        ? `
      @media (min-width: 992px) {
        grid-column: ${
          lgOffset > 0 ? `${lgOffset} / span ${lg}` : `span ${lg}`
        };
      }
    `
        : ''
    }
  `}
`;

const VericalAlignWrapper = styled.div`
  flex: 1;
`;

export interface IColProps {
  children: ReactNode;
  align?: 'centre' | 'left' | 'right';
  lg?: number;
  lgOffset?: number;
  md?: number;
  mdOffset?: number;
  sm?: number;
  smOffset?: number;
  verticalAlign?: 'middle';
  xs?: number;
  xsOffset?: number;
}

const Col: FC<IColProps> = ({
  children,
  align = 'left',
  lg = 0,
  lgOffset = 0,
  md = 0,
  mdOffset = 0,
  sm = 0,
  smOffset = 0,
  verticalAlign = null,
  xs = 12,
  xsOffset = 0,
}) => (
  <ColContainer
    align={align}
    lg={lg}
    lgOffset={lgOffset}
    md={md}
    mdOffset={mdOffset}
    sm={sm}
    smOffset={smOffset}
    verticalAlign={!!verticalAlign}
    xs={xs}
    xsOffset={xsOffset}
  >
    {verticalAlign ? (
      <VericalAlignWrapper>{children}</VericalAlignWrapper>
    ) : (
      children
    )}
  </ColContainer>
);

export default memo(Col);
