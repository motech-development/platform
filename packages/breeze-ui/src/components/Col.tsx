import { FC, ReactNode } from 'react';

export interface IColProps {
  children: ReactNode;
  align?: 'centre' | 'left' | 'right';
  lg?: number;
  lgOffset?: number;
  md?: number;
  mdOffset?: number;
  sm?: number;
  smOffset?: number;
  verticalAlign?: 'middle';
  xs?: number;
  xsOffset?: number;
}

// {
//   children,
//   align = 'left',
//   lg = 0,
//   lgOffset = 0,
//   md = 0,
//   mdOffset = 0,
//   sm = 0,
//   smOffset = 0,
//   verticalAlign = null,
//   xs = 12,
//   xsOffset = 0,
// }
const Col: FC<IColProps> = () => <div />;

export default Col;
