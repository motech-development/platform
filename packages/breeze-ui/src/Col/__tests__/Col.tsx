import { render } from '@testing-library/react';
import Col from '../Col';

describe('Col', () => {
  it('should render content', async () => {
    const { findByTestId } = render(
      <Col>
        <span data-testid="content">Hello</span>
      </Col>,
    );

    await expect(findByTestId('content')).resolves.toBeDefined();
  });

  describe('xs', () => {
    it('should have the correct default styles', () => {
      const { container } = render(<Col>Hello</Col>);

      expect(container.firstChild).toHaveStyle(`
        grid-column: span 12;
        position: relative;
      `);
    });

    it('should have the correct styles when size is set', () => {
      const { container } = render(<Col xs={6}>Hello</Col>);

      expect(container.firstChild).toHaveStyle('grid-column: span 6;');
    });

    it('should have the correct styles when size is set', () => {
      const { container } = render(
        <Col xs={6} xsOffset={4}>
          Hello
        </Col>,
      );

      expect(container.firstChild).toHaveStyle('grid-column: 4 / span 6;');
    });
  });

  describe('sm', () => {
    it('should have the correct styles when size is set', () => {
      const { container } = render(<Col sm={6}>Hello</Col>);

      expect(container.firstChild).toMatchSnapshot();
    });

    it('should have the correct styles when size is set', () => {
      const { container } = render(
        <Col sm={6} smOffset={4}>
          Hello
        </Col>,
      );

      expect(container.firstChild).toMatchSnapshot();
    });
  });

  describe('md', () => {
    it('should have the correct styles when size is set', () => {
      const { container } = render(<Col md={6}>Hello</Col>);

      expect(container.firstChild).toMatchSnapshot();
    });

    it('should have the correct styles when size is set', () => {
      const { container } = render(
        <Col md={6} mdOffset={4}>
          Hello
        </Col>,
      );

      expect(container.firstChild).toMatchSnapshot();
    });
  });

  describe('lg', () => {
    it('should have the correct styles when size is set', () => {
      const { container } = render(<Col lg={6}>Hello</Col>);

      expect(container.firstChild).toMatchSnapshot();
    });

    it('should have the correct styles when size is set', () => {
      const { container } = render(
        <Col lg={6} lgOffset={4}>
          Hello
        </Col>,
      );

      expect(container.firstChild).toMatchSnapshot();
    });
  });

  describe('alignment', () => {
    it('should have the correct styles when alignment set to left', () => {
      const { container } = render(<Col align="left">Hello</Col>);

      expect(container.firstChild).toMatchSnapshot();
    });

    it('should have the correct styles when alignment set to right', () => {
      const { container } = render(<Col align="right">Hello</Col>);

      expect(container.firstChild).toMatchSnapshot();
    });

    it('should have the correct styles when alignment set to centre', () => {
      const { container } = render(<Col align="centre">Hello</Col>);

      expect(container.firstChild).toMatchSnapshot();
    });

    it('should have the correct styles when vertical alignment set to middle', () => {
      const { container } = render(<Col verticalAlign="middle">Hello</Col>);

      expect(container.firstChild).toMatchSnapshot();
    });
  });
});
