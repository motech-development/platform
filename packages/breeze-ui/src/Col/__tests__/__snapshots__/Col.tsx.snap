// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Col alignment should have the correct styles when alignment set to centre 1`] = `
.c0 {
  grid-column: span 12;
  position: relative;
  text-align: center;
}

<div
  class="c0"
>
  Hello
</div>
`;

exports[`Col alignment should have the correct styles when alignment set to left 1`] = `
.c0 {
  grid-column: span 12;
  position: relative;
  text-align: left;
}

<div
  class="c0"
>
  Hello
</div>
`;

exports[`Col alignment should have the correct styles when alignment set to right 1`] = `
.c0 {
  grid-column: span 12;
  position: relative;
  text-align: right;
}

<div
  class="c0"
>
  Hello
</div>
`;

exports[`Col alignment should have the correct styles when vertical alignment set to middle 1`] = `
.c0 {
  grid-column: span 12;
  position: relative;
  text-align: left;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c1 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    Hello
  </div>
</div>
`;

exports[`Col lg should have the correct styles when size is set 1`] = `
.c0 {
  grid-column: span 12;
  position: relative;
  text-align: left;
}

@media (min-width:992px) {
  .c0 {
    grid-column: span 6;
  }
}

<div
  class="c0"
>
  Hello
</div>
`;

exports[`Col lg should have the correct styles when size is set 2`] = `
.c0 {
  grid-column: span 12;
  position: relative;
  text-align: left;
}

@media (min-width:992px) {
  .c0 {
    grid-column: 4 / span 6;
  }
}

<div
  class="c0"
>
  Hello
</div>
`;

exports[`Col md should have the correct styles when size is set 1`] = `
.c0 {
  grid-column: span 12;
  position: relative;
  text-align: left;
}

@media (min-width:768px) {
  .c0 {
    grid-column: span 6;
  }
}

<div
  class="c0"
>
  Hello
</div>
`;

exports[`Col md should have the correct styles when size is set 2`] = `
.c0 {
  grid-column: span 12;
  position: relative;
  text-align: left;
}

@media (min-width:768px) {
  .c0 {
    grid-column: 4 / span 6;
  }
}

<div
  class="c0"
>
  Hello
</div>
`;

exports[`Col sm should have the correct styles when size is set 1`] = `
.c0 {
  grid-column: span 12;
  position: relative;
  text-align: left;
}

@media (min-width:576px) {
  .c0 {
    grid-column: span 6;
  }
}

<div
  class="c0"
>
  Hello
</div>
`;

exports[`Col sm should have the correct styles when size is set 2`] = `
.c0 {
  grid-column: span 12;
  position: relative;
  text-align: left;
}

@media (min-width:576px) {
  .c0 {
    grid-column: 4 / span 6;
  }
}

<div
  class="c0"
>
  Hello
</div>
`;
